Triangle Streaming Server/ReceiveStream.cs

```diff
@@ -1,17 +1,53 @@
 ﻿using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Encodings;
+using Org.BouncyCastle.Crypto.Engines;
 using Org.BouncyCastle.OpenSsl;
+using Org.BouncyCastle.Security;
 using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Linq;
 using WebSocketSharp;
 using WebSocketSharp.Server;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace Triangle_Streaming_Server
 {
 	public class ReceiveStream : WebSocketBehavior
 	{
 		private static AsymmetricCipherKeyPair _keyPair;
+		private static SHA1CryptoServiceProvider _sha1;
 
-		private string _pubKey;
+		private static string _pubKey;
+
+		//A dictionary which keeps track of which clients are watching which streams.
+		//Key is client session ID, value is stream ID.
+		//TODO: optimize. Searching on value is O(n).
+		public static Dictionary<string, string> Clients { get; private set; }
+
+		public static void Initialize()
+		{
+			_sha1 = new SHA1CryptoServiceProvider();
+
+			//Read privatekey.pem from executable directory
+			using (var reader = File.OpenText(AppDomain.CurrentDomain.BaseDirectory + @"\privatekey.pem"))
```

Comment:
Which part are you referring to? The SHA1 part, or the name of the privatekey file?